### PR TITLE
[#22] Split 'transfer' into 'transfer' and 'seize'

### LINKED ISSUE #22: Split `transfer` into `transfer` and `seize`
# Clarification and motivation

Currently we have `transfer` entrypoint that can be called only by admin. It turned out that customers want something different:
1. `transfer` that works as in `ManagedLedger`: basically what we have now, but without `onlyAdmin` requirement.
2. `seize` that forcefully transfers funds and can be called only by admin.

# Acceptance criteria

1. We have `transfer` and `seize` as described above.
2. `requirements.md` is updated.
3. Tests from `morley-ledgers-test` are incorporated.

### PR DESCRIPTION
**Pull request by rvem**
## Description
`transfer` entrypoint was split into `transfer` (which is now callable by anyone) and `seize` 
(which is callable only by admin).

Since `transfer` became callable by anyone, ApprovableLedger interface tests from `morley-ledgers-test` were added.

~`seize` currently doesn't consume allowance, however, there is a small chance that it can be changed.~

## Related issue(s)

Resolves #22 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: test/Test/Indigo/Contracts/Holdings.hs [hunk 1]
```diff
@@ -284,7 +288,7 @@ test_transfer = testGroup "Test Transfer entrypoint"
         Transfer (#from .! ownerAddress, #to .! genesisAddress1, #value .! 70)
       validate . Left $
         lExpectCustomError_ #assertionFailure
-  , testCase "Transfer more tokens that have" $
+  , testCase "Transfer more tokens that have fails" $
```
Comment (gromakovsky) [inline test/Test/Indigo/Contracts/Holdings.hs:287]:
```suggestion
  , testCase "Transfer of more tokens than we have fails" $
```

File: test/Test/Indigo/Contracts/Holdings.hs [hunk 2]
```diff
@@ -293,7 +297,7 @@ test_transfer = testGroup "Test Transfer entrypoint"
         Transfer (#from .! ownerAddress, #to .! genesisAddress1, #value .! 150)
       validate . Left $
         lExpectCustomError #notEnoughBalance (#required .! 150, #present .! 100)
-  , testCase "Succesful tranfer other tokens" $
+  , testCase "Succesful transfer other's tokens" $
```
Comment (gromakovsky) [inline test/Test/Indigo/Contracts/Holdings.hs:296]:
```suggestion
  , testCase "Successful transfer of other's tokens" $
```

File: test/Test/Indigo/Contracts/Holdings.hs [hunk 3]
```diff
@@ -308,7 +312,7 @@ test_transfer = testGroup "Test Transfer entrypoint"
       lCallDef h $ GetBalance (mkView (#owner .! genesisAddress1) consumer)
       validate . Right $
         lExpectViewConsumerStorage consumer [70, 30]
-  , testCase "Transfer more tokens than approved" $
+  , testCase "Transfer fails more tokens than approved" $
```
Comment (gromakovsky) [inline test/Test/Indigo/Contracts/Holdings.hs:311]:
```suggestion
  , testCase "Transfer of more tokens than approved fails" $
```

**Comment by rvem** [comment]
> Did you run nettests?

Yes

**Pull request merged**